simmechanics/CMakeLists.txt

```diff
@@ -345,7 +345,59 @@ macro(generate_icub_simmechanics)
     list(APPEND model-simmechanics-generated-models "${CMAKE_BINARY_DIR}/${BUILD_PREFIX}/robots/${GIVTWO_YARP_ROBOT_NAME}/model.urdf")
 endmacro()
 
+macro(generate_components_simmechanics)
+    set(options INCREASE_INERTIA_FOR_GAZEBO)
+    set(oneValueArgs YARP_COMPONENT_NAME SIMMECHANICS_XML YAML_TEMPLATE CSV_TEMPLATE)
+    set(multiValueArgs)
+    cmake_parse_arguments(GIVTWO "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    #simmechanics_to_urdf SIM_LEFT_HAND_Mk3.xml --yaml simmechanics2urdf_configfile.yaml --csv-joints simmechanics2urdf_joints_configfile.csv --output xml --outputfile left_hand_mk3.urdf
+    #gz sdf -p left_hand_mk3.urdf > left_hand_mk3.sdf
+    set(GENERATED_YAML_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/${GIVTWO_YARP_COMPONENT_NAME}.yaml)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/data/${GIVTWO_YAML_TEMPLATE} ${GENERATED_YAML_LOCATION} @ONLY)    
+    set(GENERATED_CSV_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/${GIVTWO_YARP_COMPONENT_NAME}.csv)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/data/${GIVTWO_CSV_TEMPLATE} ${GENERATED_CSV_LOCATION} @ONLY)
+
+    add_custom_command(OUTPUT ${GIVTWO_YARP_COMPONENT_NAME}.urdf
+                       COMMAND simmechanics_to_urdf
+                       ARGS ${CMAKE_CURRENT_SOURCE_DIR}/data/${GIVTWO_SIMMECHANICS_XML}
+                            --output xml
+                            --yaml ${GENERATED_YAML_LOCATION}
+                            --csv-joints ${GENERATED_CSV_LOCATION}
+                            --outputfile ${GIVTWO_YARP_COMPONENT_NAME}.urdf
+                       MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/data/${GIVTWO_SIMMECHANICS_XML}"
+                       #DEPENDS  "${GENERATED_YAML_LOCATION}"
+                       #         "${GENERATED_CSV_LOCATION}"
+                       )
+
+    add_custom_command(OUTPUT "${GENERATED_YAML_LOCATION}"
+                       COMMAND ${CMAKE_COMMAND} -E 
+                               copy "${CMAKE_CURRENT_SOURCE_DIR}/data/${GIVTWO_YAML_TEMPLATE}" "${GENERATED_YAML_LOCATION}")
+
+	add_custom_command(OUTPUT "${GENERATED_CSV_LOCATION}"
```

Comment:
As a general comment, it would be great if we could avoid mixing tabs and spaces, as this induces visualization artifacts and misalignments.